android/src/main/java/com/github/yamill/orientation/listeners/OrientationListener.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.yamill.orientation.listeners
+
+import android.app.Activity
+import android.util.Log
+import android.view.OrientationEventListener
+import com.facebook.common.logging.FLog
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.facebook.react.bridge.LifecycleEventListener
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.common.ReactConstants
+import com.github.yamill.orientation.throttleLatest
+import kotlinx.coroutines.GlobalScope
+
+class OrientationListener internal constructor(
+        private val reactContext: ReactContext,
+        private val onGetCurrentActivity: () -> Activity?,
+) : LifecycleEventListener {
+
+    private lateinit var orientationEventListener: OrientationEventListener;
+
+    override fun onHostResume() {
+        val activity = onGetCurrentActivity()
+        if (activity != null && !this::orientationEventListener.isInitialized) {
+            orientationEventListener = object: OrientationEventListener(activity) {
+                override fun onOrientationChanged(orientationDegrees: Int) {
```

Comment:
This callback is very spammy, so I both
- throttled it to one event per second
- applied "distinct until changed" logic, so it won't emit the same degrees over and over again